csharp/src/Apache.Arrow.Flight.Sql/Client/FlightSqlClient.cs

```diff
@@ -0,0 +1,1211 @@
+    /// <returns>The FlightInfo describing where to access the dataset.</returns>
+    public async Task<FlightInfo> GetDbSchemasAsync(FlightCallOptions options, string? catalog = null,
+        string? dbSchemaFilterPattern = null)
+    {
+        if (options == null)
+        {
+            throw new ArgumentNullException(nameof(options));
+        }
+
+        try
+        {
+            var command = new CommandGetDbSchemas();
+
+            if (catalog != null)
+            {
+                command.Catalog = catalog;
+            }
+
+            if (dbSchemaFilterPattern != null)
+            {
+                command.DbSchemaFilterPattern = dbSchemaFilterPattern;
+            }
+
+            byte[] serializedAndPackedCommand = command.PackAndSerialize();
+            var descriptor = FlightDescriptor.CreateCommandDescriptor(serializedAndPackedCommand);
+            var flightInfoCall = GetFlightInfoAsync(options, descriptor);
+            var flightInfo = await flightInfoCall.ConfigureAwait(false);
+
+            return flightInfo;
+        }
+        catch (RpcException ex)
+        {
+            throw new InvalidOperationException("Failed to get database schemas", ex);
+        }
+    }
+
+    /// <summary>
+    /// Get the database schemas schema from the server.
+    /// </summary>
+    /// <param name="options">RPC-layer hints for this call.</param>
+    /// <returns>The SchemaResult describing the schema of the database schemas.</returns>
+    public async Task<Schema> GetDbSchemasSchemaAsync(FlightCallOptions options)
+    {
+        if (options == null)
+        {
+            throw new ArgumentNullException(nameof(options));
+        }
+
+        try
+        {
+            var command = new CommandGetDbSchemas();
+            var descriptor = FlightDescriptor.CreateCommandDescriptor(command.PackAndSerialize());
+
+            var schemaResult = await GetSchemaAsync(options, descriptor);
+            return schemaResult;
+        }
+        catch (RpcException ex)
+        {
+            throw new InvalidOperationException("Failed to get database schemas schema", ex);
+        }
+    }
+
+    /// <summary>
+    /// Given a flight ticket and schema, request to be sent the stream. Returns record batch stream reader.
+    /// </summary>
+    /// <param name="options">Per-RPC options</param>
+    /// <param name="ticket">The flight ticket to use</param>
+    /// <returns>The returned RecordBatchReader</returns>
+    public async IAsyncEnumerable<RecordBatch> DoGetAsync(FlightCallOptions options, FlightTicket ticket)
+    {
+        if (options == null)
+        {
+            throw new ArgumentNullException(nameof(options));
+        }
+
+        if (ticket == null)
+        {
+            throw new ArgumentNullException(nameof(ticket));
+        }
+
+        var call = _client.GetStream(ticket, options.Headers);
+        await foreach (var recordBatch in call.ResponseStream.ReadAllAsync())
```

Comment:
Add `ConfigureAwait(false)` if supported (I've not tried on a foreach)

Reply:
done